src/target/stm32f1.c

```diff
@@ -254,6 +258,87 @@ bool at32fxx_probe(target_s *t)
 	return false;
 }
 
+/* Identify MM32 devices (Cortex-M0) */
+bool mm32l0xx_probe(target_s *t)
+{
+	uint32_t mm32_id;
+	const char *name = "?";
+	size_t flash_kbyte = 0;
+	size_t ram_kbyte = 0;
+	size_t block_size = 0x400;
+
+	if (!t)
+		return false;
+	if (target_mem_read(t, &mm32_id, DBGMCU_IDCODE_MM32L0, sizeof(mm32_id))) {
+		DEBUG_WARN("mm32xx_probe: read error at 0x%" PRIx32 "\n", (uint32_t)DBGMCU_IDCODE_MM32L0);
+		return false;
+	}
+	switch (mm32_id) {
+	case 0xcc568091U:
+		name = "MM32L07x";
+		flash_kbyte = 128;
+		ram_kbyte = 8;
+		break;
+	case 0xcc56a097U:
+		name = "MM32SPIN27";
+		flash_kbyte = 128;
+		ram_kbyte = 12;
+		break;
+	case 0x0:
+	case 0xffffffff:
+		return false;
+	default:
+		DEBUG_WARN("mm32_probe: unknown mm32 dev_id 0x%" PRIx32 "\n", mm32_id);
+		return false;
+	}
+	t->part_id = mm32_id & 0xfff;
+	t->driver = name;
+	t->mass_erase = stm32f1_mass_erase;
+	target_add_ram(t, 0x20000000, ram_kbyte * 1024U);
+	stm32f1_add_flash(t, 0x8000000, flash_kbyte * 1024U, block_size);
+	target_add_commands(t, stm32f1_cmd_list, name);
+	if (cortexm_ap(t) && cortexm_ap(t)->dp)
```

Comment:
Turn all use of `cortexm_ap(t)` here into use of a local. This is hard to reason about and needlessly pessimises performance on debug builds. It's also not possible for the adiv5_access_port_t structure pointer to be NULL here, nor the DP one. These are guarantees from the adiv5 driver. Remove the extra checks.